pkg/k8sexporter/k8s_exporter.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sexporter
+
+import (
+	"net"
+	"net/http"
+	_ "net/http/pprof"
+	"strconv"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"k8s.io/node-problem-detector/cmd/options"
+	"k8s.io/node-problem-detector/pkg/condition"
+	"k8s.io/node-problem-detector/pkg/problemclient"
+	"k8s.io/node-problem-detector/pkg/types"
+	"k8s.io/node-problem-detector/pkg/util"
+)
+
+type k8sExporter struct {
+	client           problemclient.Client
+	conditionManager condition.ConditionManager
+}
+
+// NewExporterOrDie creates a exporter for Kubernetes apiserver exporting, panics if error occurs.
+func NewExporterOrDie(npdo *options.NodeProblemDetectorOptions) types.Exporter {
+	if !npdo.EnableK8sExporter {
+		return nil
+	}
+
+	c := problemclient.NewClientOrDie(npdo)
```

Comment:
problemclient package should be renamed. That is just for k8s now. In terms of directory structure. Create `/pkg/exporters`. Put k8s and prometheus exporters into that.

Reply:
Good point. Done.